cmd/cloud-node-manager/app/nodemanager.go

```diff
@@ -34,126 +30,95 @@ import (
-	s, err := options.NewCloudNodeManagerOptions()
-	if err != nil {
-		klog.Fatalf("unable to initialize command options: %v", err)
-	}
-
-	cmd := &cobra.Command{
-		Use:  "cloud-node-manager",
-		Long: `The Cloud node manager is a daemon that reconciles node information for its running node.`,
-		Run: func(cmd *cobra.Command, args []string) {
-			verflag.PrintAndExitIfRequested("Cloud Node Manager")
-			cliflag.PrintFlags(cmd.Flags())
-
-			c, err := s.Config()
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
-			}
-
-			if err := initForOS(c.WindowsService); err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
-			}
-
-			if err := Run(c, wait.NeverStop); err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
-			}
-
-		},
-	}
+var cloudNodeManagerOptions = options.NewCloudNodeManagerOptions()
 
-	fs := cmd.Flags()
-	namedFlagSets := s.Flags()
+func init() {
+	fs := Rootcmd.Flags()
+	namedFlagSets := cloudNodeManagerOptions.Flags()
 	verflag.AddFlags(namedFlagSets.FlagSet("global"))
-	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
+	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), Rootcmd.Name())
 	for _, f := range namedFlagSets.FlagSets {
 		fs.AddFlagSet(f)
 	}
+
 	usageFmt := "Usage:\n  %s\n"
-	cols, _, _ := term.TerminalSize(cmd.OutOrStdout())
-	cmd.SetUsageFunc(func(cmd *cobra.Command) error {
+	cols, _, _ := term.TerminalSize(Rootcmd.OutOrStdout())
+	Rootcmd.SetUsageFunc(func(cmd *cobra.Command) error {
 		fmt.Fprintf(cmd.OutOrStderr(), usageFmt, cmd.UseLine())
-		cliflag.PrintSections(cmd.OutOrStderr(), namedFlagSets, cols)
+		cliflag.PrintSections(Rootcmd.OutOrStderr(), namedFlagSets, cols)
 		return nil
 	})
-	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+	Rootcmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
 		fmt.Fprintf(cmd.OutOrStdout(), "%s\n\n"+usageFmt, cmd.Long, cmd.UseLine())
 		cliflag.PrintSections(cmd.OutOrStdout(), namedFlagSets, cols)
 	})
-
-	return cmd
 }
 
-// Run runs the ExternalCMServer.  This should never exit.
-func Run(c *cloudnodeconfig.Config, stopCh <-chan struct{}) error {
-	// To help debugging, immediately log version and nodeName
-	klog.Infof("Version: %+v", version.Get())
-	klog.Infof("NodeName: %s", c.NodeName)
-
-	// Start the controller manager HTTP server
-	var checks []healthz.HealthChecker
-	healthzHandler := controllerhealthz.NewMutableHealthzHandler(checks...)
-	if c.SecureServing != nil {
-		unsecuredMux := genericcontrollermanager.NewBaseHandler(&config.DebuggingConfiguration{}, healthzHandler)
-		handler := genericcontrollermanager.BuildHandlerChain(unsecuredMux, &c.Authorization, &c.Authentication)
-		// TODO: handle stoppedCh returned by c.SecureServing.Serve
-		if _, _, err := c.SecureServing.Serve(handler, 0, stopCh); err != nil {
+// NewCloudNodeManagerCommand creates a *cobra.Command object with default parameters
+var Rootcmd = &cobra.Command{
```

Comment:
What is the benefit of this rather than Run function? I am not sure whether it is worth a refactor here.

Reply:
Just follow the cobra's example. It is easier to get config value. There will be only one cmd instance. It's ok to define a global variable.